client-cli/src/main/java/au/org/consumerdatastandards/client/cli/support/ApiUtil.java

```diff
@@ -103,27 +111,24 @@ public static ApiClient createApiClient(ApiClientOptions clientOptions) throws A
             LOGGER.info("User Agent is set to {}", userAgent);
         }
 
-        String accessToken = clientOptions.getAccessToken();
-        if (StringUtils.isNotBlank(accessToken) || StringUtils.isNotBlank(clientOptions.getRefreshToken())) {
-            Integer exp = null;
-            Map<String, Object> claims = null;
-            if (StringUtils.isNotBlank(accessToken)) {
+        if (authRequired) {
+            if ((accessTokenExpiresAt == 0 || accessTokenExpiresAt < System.currentTimeMillis() + EXPIRY_BUFFER)
+                    && StringUtils.isNotBlank(clientOptions.getRefreshToken())) {
+                renewTokens(clientOptions, originalHttpClient);
+            }
+            if (StringUtils.isBlank(clientOptions.getAccessToken()) && StringUtils.isNotBlank(clientOptions.getAuthServer())) {
                 try {
-                    claims = parseClaims(accessToken);
-                    exp = (Integer) claims.get("exp");
-                } catch (ApiException e) {
-                    LOGGER.info("Invalid access token.");
+                    launchBrowser("http://localhost:8686/auth");
```

Comment:
perhaps get the port from the properties instead of hard coding here?

Reply:
Good idea. Done. Should note however that the port can't be completely random - must be known and configured on the auth server as part of the registered redirect URL for the client.